tests/typing/client_typing.py

```diff
@@ -0,0 +1,21 @@
+import io
+
+from clickhouse_connect.driver import Client
+from typing import assert_type, cast, Union
+
+# Test Client.raw_query overloads:
+
+client = cast(Client, None)
+
+result = client.raw_query("")
+assert_type(result, bytes)
+
+result = client.raw_query("", stream=False)
+assert_type(result, bytes)
+
+result = client.raw_query("", stream=True)
+assert_type(result, io.IOBase)
+
+stream = cast(bool, None)
+result = client.raw_query("", stream=stream)
+assert_type(result, Union[bytes, io.IOBase])
```

Comment:
See PR description:

> I manually ran `pyright tests/typing/client_typing.py` and observed there were no errors. Ideally, running a type checker would be part of this project's CI set up, but this is out of scope for this PR.

Would you be open to a PR that adds a CI job to run [pyright](https://github.com/microsoft/pyright) on the code base?

Reply:
I don't see any reason not to?  So I think that a PR for that makes sense.

Reply:
In this particular case maybe some refactoring makes sense to separate a `stream` response (which is only used by arrow) from a regular bytes response, instead of making it an overload.  The overload approach seems to add a lot of repetition.  I'll give it some more thought this week.

Reply:
> In this particular case maybe some refactoring makes sense to separate a stream response (which is only used by arrow) from a regular bytes response, instead of making it an overload.

Ah, I guess you are considering the “stream” parameter to not be part of the public API then?

Reply:
At this point no, I'd rather have a wrapper or some other method of signaling that the method returns a stream vs a bunch of bytes instead of an optional parameter that changes the return type.  I didn't really give it a lot of thought when considering the community PR that made the change.